pkg/services/pluginsintegration/keystore/keystore.go

```diff
@@ -0,0 +1,72 @@
+package keystore
+
+import (
+	"context"
+	"time"
+
+	"github.com/grafana/grafana/pkg/infra/kvstore"
+	"github.com/grafana/grafana/pkg/plugins/log"
+)
+
+// Service is a service for storing and retrieving public keys.
+type Service struct {
+	kv   *kvstore.NamespacedKVStore
+	slog log.Logger
+}
+
+const (
+	prefix         = "key-"
+	lastUpdatedKey = "last_updated"
+)
+
+func ProvideService(kv kvstore.KVStore) *Service {
+	return &Service{
+		kv:   kvstore.WithNamespace(kv, 0, "plugin.publickeys"),
+		slog: log.New("plugin.publickeys"),
+	}
+}
+
+func (s *Service) Get(ctx context.Context, key string) (string, bool, error) {
+	return s.kv.Get(ctx, prefix+key)
+}
+
+func (s *Service) Set(ctx context.Context, key string, value string) error {
+	return s.kv.Set(ctx, prefix+key, value)
+}
+
+func (s *Service) Del(ctx context.Context, key string) error {
+	return s.kv.Del(ctx, prefix+key)
+}
+
+func (s *Service) GetLastUpdated(ctx context.Context) time.Time {
+	lastUpdated := time.Time{}
+	if val, ok, err := s.kv.Get(ctx, lastUpdatedKey); err != nil {
+		s.slog.Error("Failed to get last sent time", "error", err)
```

Comment:
Nit: I would return an error in this function rather than logging, so the caller can distinguish between an underlying failure and "job never run"

```go
lastUpdated := time.Time{}
val, ok, err := s.kv.Get(ctx, lastUpdatedKey)
if err != nil {
	return lastUpdated, fmt.Errorf("get: %w")
}
if !ok {
	// return zero-value time.Time
	return lastUpdated, nil
}
```
And I would probably treat the `time.Parse` failure as when the key does not exist (`return time.Time{}, nil`)

But interested in hearing what other folks think as well :grinning: